### ISSUE TITLE
Error while using Regex on queries

### ISSUE THREAD

**Issue by jlanza** [ScorpioBroker/ScorpioBroker#398]
Hi,

I'm sorry to inform about another issue. Using regex in queries was working in a former version, but upgrading to the latest docker version of Scorpio and PostgreSQL (15) I'm facing some errors. I'm using the id filtering as an example, but you can use whatever other parameter.

```
GET http://myscorpio.domain.com/ngsi-ld/v1/entities?type=MyType&q=id~=.*test.*
Accept: application/ld+json
```
I get the following error:

```
HTTP/1.1 500 Internal Server Error
content-length: 180
connection: close
Content-Type: application/json

{
  "type": "https://uri.etsi.org/ngsi-ld/errors/InternalError",
  "title": "Internal error",
  "detail": {
    "message": "ERROR: operator does not exist: jsonb ~ unknown (42883)"
  },
  "errorCode": 500
}
```

Scorpio logs reports:
```
2023-06-22 14:49:16,567 ERROR [eu.nec.ngs.com.too.HttpUtils] (vert.x-eventloop-thread-71) Exception :: : io.vertx.pgclient.PgException: ERROR: operator does not exist: jsonb ~ unknown (42883)
	at io.vertx.pgclient.impl.codec.ErrorResponse.toException(ErrorResponse.java:31)
	at io.vertx.pgclient.impl.codec.PrepareStatementCommandCodec.handleErrorResponse(PrepareStatementCommandCodec.java:90)
	at io.vertx.pgclient.impl.codec.PgDecoder.decodeError(PgDecoder.java:246)
	at io.vertx.pgclient.impl.codec.PgDecoder.decodeMessage(PgDecoder.java:132)
	at io.vertx.pgclient.impl.codec.PgDecoder.channelRead(PgDecoder.java:112)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800)
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:499)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:397)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

As I said when using PostgreSQL 13 it seems to be working.

Thanks for the great work carried out.

**Comment by ScorpioBroker** [comment]
Hi we are currently checking this out ... as a workaround for now (well it is actually the recommended way) please use idPattern=.\*test.\*

**Comment by ScorpioBroker** [comment]
fixed in 4.0.3
